beerus_rest_api/src/api/starknet/endpoints.rs

```diff
@@ -279,6 +279,33 @@ pub async fn query_starknet_syncing(
     ApiResponse::from_result(query_syncing_inner(beerus).await)
 }
 
+/// Query the estimated gas fees for a StarkNet Transaction.
+
+/// The estimated gas fees.
+///
+/// # Arguments
+///
+/// * `block_id_type` - Type of block identifier. eg. hash, number, tag
+/// * `block_id` - The block identifier. eg. 0x123, 123, pending, or latest
+/// * `broadcasted_transaction` - The broadcasted transaction to be estimated (URLencoded)
+///
+/// # Returns
+///
+/// `Ok(FeeEstimate)` if the operation was successful.
+/// `Err(eyre::Report)` if the operation failed.
+#[openapi]
+#[get("/starknet/fee?<broadcasted_transaction>&<block_id>&<block_id_type>")]
+pub async fn get_estimate_fee(
+    beerus: &State<BeerusLightClient>,
+    block_id_type: String,
+    block_id: String,
+    broadcasted_transaction: String,
+) -> ApiResponse<QueryEstimateFeeResponse> {
+    ApiResponse::from_result(
+        get_estimate_fee_inner(beerus, block_id_type, block_id, broadcasted_transaction).await,
+    )
+}
```

Comment:
This is what the call looks like, `broadcasted_transaction` is JSON URL-encoded

```bash
curl --silent \
http://127.0.0.1:8000/starknet/fee\?broadcasted_transaction\=%7B%22type%22%3A%22INVOKE%22%2C%22max_fee%22%3A%220x0%22%2C%22version%22%3A%220x1%22%2C%22signature%22%3A%5B%220x156a781f12e8743bd07e20a4484154fd0baccee95d9ea791c121c916ad44ee0%22%2C%220x7228267473c670cbb86a644f8696973db978c51acde19431d3f1f8f100794c6%22%5D%2C%22nonce%22%3A%220x0%22%2C%22sender_address%22%3A%220x5b5e9f6f6fb7d2647d81a8b2c2b99cbc9cc9d03d705576d7061812324dca5c0%22%2C%22calldata%22%3A%5B%220x1%22%2C%220x7394cbe418daa16e42b87ba67372d4ab4a5df0b05c6e554d158458ce245bc10%22%2C%220x2f0b3c5710379609eb5495f1ecd348cb28167711b73609fe565a72734550354%22%2C%220x0%22%2C%220x3%22%2C%220x3%22%2C%220x5b5e9f6f6fb7d2647d81a8b2c2b99cbc9cc9d03d705576d7061812324dca5c0%22%2C%220x3635c9adc5dea00000%22%2C%220x0%22%5D%7D\&block_id\=latest\&block_id_type\=tag | jq
```

This is the output
```json
{
  "gas_consumed": "5194",
  "gas_price": "22630904152",
  "overall_fee": "117544916165488"
}
```

Reply:
tested looks great